task_processing/plugins/mesos/execution_framework.py

```diff
@@ -483,19 +483,20 @@ def statusUpdate(self, driver, update):
         )
 
         # Record state changes
-        if md.task_state != update.state:
+        task_state = update.state.encode('utf-8')
```

Comment:
Why do we want bytes here? If there is a validator just change it to str and it should work ok? (or call str on update.state)